### PR TITLE
improve(relayer): Drop slow fill request queries

### PR DESCRIPTION
**Pull request by pxrl**
By retaining the deposit FillStatus on the destination chain it's possible to skip scraping for RequestedV3SpeedUp events. This reduces the RPC load during SpokePoolClient update.

### CODE REVIEW

File: src/relayer/Relayer.ts [hunk 1]
```diff
@@ -262,17 +267,13 @@ export class Relayer {
...
-        this.logger.debug({
-          at: "Relayer",
-          message: "Initiating slow fill for grey listed depositor",
-          depositor,
-        });
-        this.requestSlowFill(deposit);
-      }
-      // Regardless of whether we should send a slow fill or not for this depositor, exit early at this point
-      // so we don't fast fill an already slow filled deposit from the slow fill-only list.
+    if (slowDepositors?.includes(depositor) && fillStatus === FillStatus.Unfilled) {
```
Comment (pxrl) [inline src/relayer/Relayer.ts:270]: Opinionated driveby change: If the operator has configured `SLOW_DEPOSITORS`, then don't require `SEND_SLOW_RELAYS` to be enabled.

File: src/relayer/Relayer.ts [hunk 2]
```diff
@@ -262,17 +267,13 @@ export class Relayer {
     }
 
     // If depositor is on the slow deposit list, then send a zero fill to initiate a slow relay and return early.
-    if (slowDepositors?.includes(depositor)) {
-      if (sendSlowRelays) {
```
Comment (pxrl) [inline src/relayer/Relayer.ts:266]: Opinionated driveby change: If the operator has configured SLOW_DEPOSITORS, then don't require SEND_SLOW_RELAYS to be enabled.

Comment (nicholaspai) [reply src/relayer/Relayer.ts:266]: why not? not sure I agree

**Review by nicholaspai — APPROVED** [review]
This also seems good

**Review by james-a-morris — COMMENTED** [review]
Although I agree that a slow depositor should trigger a slow relay every time, I'm not sure we want to remove the configuration to be able to disable slow relays from a given relayer.

**Comment by pxrl** [comment]
> Although I agree that a slow depositor should trigger a slow relay every time, I'm not sure we want to remove the configuration to be able to disable slow relays from a given relayer.

Agree - that's not changed here and it remains as-is :+1: Slow relays is still 100% opt-in.

File: src/relayer/Relayer.ts (hunk 2 above)
Comment (pxrl) [reply src/relayer/Relayer.ts:267]: The current configuration gives no way to opt-in to slow fill specific addresses and to ignore all others. Meanwhile, if the operator has explicitly defined an array of addresses to fill with slow relays, I see that as an implicit opt-in to slow relays for those addresses, and it shouldn't be required to separately enable slow fills for all other addresses.

This doesn't mean that the relayer would make slow fills for all other deposits - only for addresses identified in `slowDepositors`. The existing `sendSlowRelays` conditional still applies to everything else.

**Review by nicholaspai — COMMENTED** [review]
I think this will also cause bundle refund construction problematic since we need slow fill requests to compute bundle data correctly

**Review by nicholaspai — CHANGES_REQUESTED** [review]
do not merge yet until bundle data can be fetched without using SlowFillRequest events

**Comment by pxrl** [comment]
> I think this will also cause bundle refund construction problematic since we need slow fill requests to compute bundle data correctly

That's important for the proposer/disputer, but is relevant for the relayer? This PR only affects the relayer's SpokePoolClient update, and deposit update requests don't affect relayer repayments so I can't see any obvious problem with ignoring them.

**Review by nicholaspai — APPROVED** [review]
I think we can revisit this now that we load most bundle data from arweave. I also think this probably wasn't an issue affecting bundle refund computation

**Comment by pxrl** [comment]
> I think we can revisit this now that we load most bundle data from arweave. I also think this probably wasn't an issue affecting bundle refund computation

Yeah - I also wasn't able to reason my way to any problem with dropping these events. I'll merge this on Monday :+1:

**Pull request merged**